From f712e42bd15b05003b122788779edf3b5333e48b Mon Sep 17 00:00:00 2001
From: WeiYong Bi <bivvy.bi@rock-chips.com>
Date: Mon, 12 Dec 2016 16:40:21 +0800
Subject: [PATCH] drm/panel: simple: Add support for AUO B125HAN03.1 1920x1080
 panel

The AUO B125HAN03.1 panel is a 12.5' FHD 16:9 Color TFT-LCD
and 30 pins eDP interface.

Change-Id: Ic2be5f1d5a3e25d805e0752c1c1fad0decc4f2d0
Signed-off-by: WeiYong Bi <bivvy.bi@rock-chips.com>
---
 .../bindings/display/panel/auo,b125han03.txt  |  7 +++++
 drivers/gpu/drm/panel/panel-simple.c          | 28 +++++++++++++++++++
 2 files changed, 35 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/display/panel/auo,b125han03.txt

diff --git a/Documentation/devicetree/bindings/display/panel/auo,b125han03.txt b/Documentation/devicetree/bindings/display/panel/auo,b125han03.txt
new file mode 100644
index 000000000000..aef90890dca4
--- /dev/null
+++ b/Documentation/devicetree/bindings/display/panel/auo,b125han03.txt
@@ -0,0 +1,7 @@
+AU Optronics Corporation 12.5" FHD TFT LCD panel
+
+Required properties:
+- compatible: should be "auo,b125han03"
+
+This binding is compatible with the simple-panel binding, which is specified
+in simple-panel.txt in this directory.
diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index 95e5710794ce..427c53008e1d 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -553,6 +553,31 @@ static const struct panel_desc auo_b116xw03 = {
 	},
 };
 
+static const struct drm_display_mode auo_b125han03_mode = {
+	.clock = 141000,
+	.hdisplay = 1920,
+	.hsync_start = 1920 + 88,
+	.hsync_end = 1920 + 88 + 60,
+	.htotal = 1920 + 88 + 60 + 36,
+	.vdisplay = 1080,
+	.vsync_start = 1080 + 12,
+	.vsync_end = 1080 + 12 + 4,
+	.vtotal = 1080 + 12 + 4 + 20,
+	.vrefresh = 60,
+	.flags = DRM_MODE_FLAG_NVSYNC | DRM_MODE_FLAG_NHSYNC,
+};
+
+static const struct panel_desc auo_b125han03 = {
+	.modes = &auo_b125han03_mode,
+	.num_modes = 1,
+	.bpc = 6,
+	.size = {
+		.width = 276,
+		.height = 156,
+	},
+	.bus_format = MEDIA_BUS_FMT_RGB666_1X18,
+};
+
 static const struct drm_display_mode auo_b133xtn01_mode = {
 	.clock = 69500,
 	.hdisplay = 1366,
@@ -1303,6 +1328,9 @@ static const struct of_device_id platform_of_match[] = {
 	}, {
 		.compatible = "auo,b116xw03",
 		.data = &auo_b116xw03,
+	}, {
+		.compatible = "auo,b125han03",
+		.data = &auo_b125han03,
 	}, {
 		.compatible = "auo,b133htn01",
 		.data = &auo_b133htn01,
-- 
2.35.3

